### PR TITLE
Fix forward AD for cudnn batch norm

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71901

~We didn't catch this initially because CuDNN is not being tested on CI.~

The following tests fail on master (if we build with CuDNN), but pass with this PR:
- `test_forward_mode_AD_nn_functional_batch_norm_cuda_float64`
- `test_forward_mode_AD_nn_functional_instance_norm_cuda_float64`

I don't think it is documented anywhere, but from the tests passing now I'm going to guess `result1` and `result2` return `mean` and `invstd` respectively. Previously, I thought mean and variance were returned because the variables were named `saved_mean` and `saved_var`.

Differential Revision: [D33818652](https://our.internmc.facebook.com/intern/diff/D33818652)

### CODE REVIEW

**Comment by soulitzer** [comment]
@anjali411 btw I still haven't been able to replicate the failures for batch/instance norm on CI https://github.com/pytorch/pytorch/runs/4945367611?check_suite_focus=true after you landed your ZT fix (when USE_CUDNN=0), though this might clear up some confounding factors.

**Review by albanD — APPROVED** [review]
👍

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33818652).
